cpp/ql/test/library-tests/ir/range-analysis/test.cpp

```diff
@@ -130,3 +130,13 @@ void test_div(int x) {
     range(x >> 2); // $ range=>=0 range=<=2
   }
 }
+
+struct X { int n; };
+void read_argument(const X *);
+
+void nonterminating_without_operands_as_ssa(X *x) {
+  read_argument(x);
+  while (x->n) {
+    x->n--;
+  }
+}
```

Comment:
There are no expected results for the test.  It might be nice to either add some and/or explain that we're mostly checking for an analysis non-termination issue here?

Reply:
Yeah, it's a little odd that we're supposed to read the function name as a comment on the implementation, though we're not exactly strict about such things.  I'd prefer a comment.  I don't have strong feelings.